app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,136 @@
+class OrdersController < ApplicationController
+  
+  before_action :authenticate_user!
+  before_action :set_journey, only: %i(new create checkout)
+  before_action :set_order, only: %i(checkout charge)
+  before_action :instantiate_order, only: %i(create)
+  before_action :ensure_booking_is_active, only: %i(new checkout create)
+  
+  def checkout
+  end
+  
+  def new
+    @minutes = calculate_minutes
+    @order = Order.new
+    authorize(@order)
+    # add here all the sum of all the minutes that have not been paid. 
+  end 
+  
+  def create
+    @order.consumer_total = @order.minutes * @order.price_hour / 60
+    @order.state = 'not paid'
+    if @order.valid?
+      authorize(@order)
+      if @order.save
+        flash[:success] = t 'orders.create.success'
+        redirect_to dashboard_path
+      else
+        flash.now[:error] = t 'orders.create.unexpected_error'
+        handle_errors
+      end
+    else
+      flash.now[:error] = t 'orders.create.validation_error'
+      authorize(@order, :true?)
+      handle_errors
+    end
+  end
+  
+  def charge
+    
+    # stripe_token = params.require :stripeToken
+    # if @order.charge! stripe_token
+    #   flash[:success] = t '.success'
+    #   redirect_to @order.reservation_link
+    # else
+    #   @order.reload # for cleaning up any possibly modified attributes
+    #   flash[:error] = t 'stripe.payment_failed'
+    #   redirect_to @order.payment_link
+    # end
+    
+    customer = Stripe::Customer.create(
+      source: params[:stripeToken],
+      email:  params[:stripeEmail]
+    )
+
+    charge = Stripe::Charge.create(
+      customer:     customer.id,   
+      amount:       @order.consumer_total_cents, 
+      description:  @order.charge_description,
+      currency:     @order.consumer_total.currency
+    )
+    
+  @order.update(payment: charge.to_json, state: 'paid')
+  flash[:success] = t '.success'
+  redirect_to dashboard_path
+    
+  rescue Stripe::CardError => e
+    flash[:error] = e.message
+    redirect_to dashboard_path
+    
```

Comment:
> For instance when I test stripe with an expired card I get: `Render and/or redirect were called multiple times in this action.`

This has nothing to do with Stripe. It's purely a Rails-level error. Focus on that first (double-render or double-redirect), and only then worry about Stripe details